packages/collection/src/collection.ts

```diff
@@ -662,6 +662,21 @@ export class Collection<K, V> extends Map<K, V> {
 		return coll;
 	}
 
+	/**
+	 * The complement method returns a new structure containing items where the keys and values of other structure are not present in the original.
+	 *
+	 * @param other - The other Collection to filter against
+	 */
+	public complement<T>(other: ReadonlyCollection<K, T>): Collection<K, T> {
+		const coll = new this.constructor[Symbol.species]<K, T>();
+		for (const [k, v] of other) {
+			if (!(this.has(k) || Object.is(v, this.get(k)))) {
```

Comment:
```suggestion
			if (!this.has(k) && !Object.is(v, this.get(k))) {
```

Reply:
Isn't my one faster?

Reply:
Shouldn't sacrifice readability for speed, except we talk like 10x or something.